codesign/codesign.go

```diff
@@ -231,6 +232,12 @@ func SelectConnectionCredentials(
 			return appleauth.Credentials{}, fmt.Errorf("failed to restore Apple ID login session: %w", err)
 		}
 
+		if session != "" &&
+			bitriseConnection.AppleIDConnection.SessionExpiryDate != nil &&
+			bitriseConnection.AppleIDConnection.SessionExpiryDate.Before(time.Now()) {
+			logger.Warnf("Two-factor session has expired at: %s", bitriseConnection.AppleIDConnection.SessionExpiryDate.Format("2006-01-02 15:04"))
```

Comment:
nit: We could also add a suggestion and a next step, like "go to your app settings (on Bitrise) and renew the connection" or a Devcenter URL.